falcon/routing/static.py

```diff
@@ -106,7 +106,11 @@ def __call__(self, req, resp):
             except IOError:
                 raise falcon.HTTPNotFound()
 
-        suffix = os.path.splitext(file_path)[1]
+        path_with_extension = file_path
```

Comment:
This solution makes sense, however a little further down, `downloadable_as` pulls the file name from the `file_path`, wouldn't we want to update that as well?

For example, if someone requests a CSS file and we fallback to an HTML file, the content type will be `text/html` however the extension will be `.css` When in a browser that isn't a big deal I wouldn't think, but once it hits a filesystem, the OS will infer it incorrectly.

When we set the `resp.stream` in the `try...except` above, we might want to set the `file_path` there as well. That would also eliminate the `if`.